auth/mock.go

```diff
@@ -0,0 +1,32 @@
+package auth
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/oauth2"
+)
+
+type MockSession struct{}
+
+// GetToken returns some mock token with static data.
+func (s *MockSession) GetToken(ctx context.Context, conf *oauth2.Config) (*oauth2.Token, error) {
+	tok := oauth2.Token{
+		AccessToken:  "SomeAccessToken",
+		RefreshToken: "SomeRefreshToken",
+		Expiry:       time.Now().Add(time.Hour * 1),
+	}
+	return &tok, nil
+}
```

Comment:
We as of currently aren't using our MockSession implementation of GetToken()  anywhere since the MockSession GetTokenFromBrowser() offers more precision and test coverage. The method is required by the Session interface, but maybe its body isn't needed here if we aren't using it anywhere? Just a knitpick, but otherwise LGTM!

Reply:
Let's keep it as is and add an assertion here to verify its use: https://github.com/hashicorp/hcp-sdk-go/pull/129/files#r1019650524